### DISCUSSION TITLE
Workflow inside workflow, nested workflow support or workflow  interconnection

### DISCUSSION THREAD

**Discussion by nmocruz** [elsa-workflows/elsa-core#508] [Q&A]
Hi,

I was wondering if there's any way to integrate workflow inside other workflows.
For example, Workflow A, controller higher-level states o Pocress A, is a long process while that, some other activities can happen and ideally this could be described and managed by other workflows B, C and D

**Comment by sfmskywalker** [comment]
Yes, Elsa 2 has support for parent/child workflows. It has a `RunWorkflow` activity that allows you to run a given workflow and configure whether you want the parent workflow to wait for the child workflow to complete, or if it should continue running ("fire-and-forget").

It also has support for composite activities, which are custom activities that are modeled the same way as a workflow, effectively allowing you to build higher-level activities in the form of smaller workflows (although they could be as complex as you like).

**Reply by nmocruz** [reply]
Seems perfect.
I'm now considering some architecture challenges to a system.  The old version has a lot of things that could be better build if was build with a workflow tool. I will try to do some experiments, my last experience with workflows was with WF 10 years ago

**Reply by nmocruz** [reply]
Other question, There's any forum, chat channel, like gitter, discord? I may have many questions, in next weeks

**Reply by nmocruz** [reply]
Ok, gonna use Github discussions, didnt saw

**Comment by Alexplose** [comment]
Currently using Elsa 1.x, and implemented a StartWorkflowActivity with a fire-and-forget behavior, that copies all parents workflow variables to child.
It's working well, let me know if it may be useful to you.
